drake/core/Path.h.in

```diff
@@ -2,7 +2,7 @@
 
 #include <string>
 
-namespace Drake {
+namespace {
```

Comment:
Emitting Path.cpp seems best to me too. There is one other option: declare the function `inline`. The language guarantees that there will only be one of those so you won't get duplicate symbol problems.